### DISCUSSION TITLE
Feature Request: Google Home Support in Tasmota

### DISCUSSION THREAD

**Discussion by Pewibe** [arendst/Tasmota#14390] [Ideas]
As I started with Tasmota I bought devices which were equipped with software allowing control by Alexa or Google Home. A quick web search discovered that there are a lot of guys controlling Tasmota devices by Google Home. Additionally Tasmota has 2 emulations which gave me to trust into a connection between Tasmota and Google Home, whenever I want it.
Now my wife asked me why we cannot use our Google Home Mini to control all the Tasmota gadgets. I said "no problem" and didn't know how far I'm away from reality. Looking for detail in the web I discovered usage of helping constructs like IFTTT or libraries for Arduino (not Tasmota), etc. A web service "g-bridge" supporting the constellation is dead in the meanwhile. Even the Tasmota doc owns a top level chapter "Smart Home Integrations" where you can find Alexa but no description for using Google Home.
So I like to ask you readers if there is someone out there to add a description for controlling Tasmota devices via Google Home - as local as possible. If somebody knows about such an existing description, please let me know and I will prepare it for the Tasmota project after I verified it in practice. Often a good drawing is much better than pages of text.
If you agree that there is nothing of value to guide someone for an integration, maybe I can invite a guy (who understands the principles) to generate the missing? Why is Alexa described and working in / with Tasmota but not Google Home? If a software / service is needed to close the gap - why there are multiple a Tasmota flashing apps or Tasmota management apps but not only one to help Google Home device owners?

**Comment by barbudor** [comment]
> Why is Alexa described and working in / with Tasmota but not Google Home?
>  why there are multiple a Tasmota flashing apps or Tasmota management apps but not only one to help Google Home device owners?

Tasmota is essentially made from volunteer contributor. There is no real Master Plan, no Marketing Department, no Roadmap.
Unless someone has the need, the will and the skills, nothing happens.
I don't have Google Home (neither Alexa and none of these assistants will ever enter my home) so I am example of the one from where this won't happen.
If this is only docs, anyone can contribute to the documentation and add more informations and guides. If you can do that, please please do.
If it needs coding, you will have to do it of find somone who is interrested and has a Google Home device.

**Reply by Pewibe** [reply]
Understood. I personally have the will, some need, but no skill to update the doc. Therefore I even do not know if it needs coding or a contract with Google, etc. 
You pointed out the target of my question: "to ... find somone who is interrested and has a Google Home device"!

**Comment by Pewibe** [comment]
In accordance to what @barbudor stated, I changed the title of this discussion. Open items:

**1. Tasmota doc chapter "Smart Home Integrations", subchapter "Google Home / Nest":** Who can add a description a) why it is not possible today and/or b) what is needed to use Google Home as integration service?
**2. Tasmota as "seamless" device for Google Home:** What has to be added to Tasmota? Who can deliver a specification? Who can apply necessary code?
**3. Tasmota as "compatible" device for Google Home:** In this case I guess a web service is necessary. Who can deliver a specification? Who can apply necessary code?

**Reply by mikimast** [reply]
Buongiorno e buon anno.
Sono trascorsi tanti giorni e mi permetto di chiedere se ci sono novità. Non capisco perchè questo problema non sia stato ancora risolto. A me non risultano possibilità di raggiungere un dispositivo TASMOTA con il nostro GOOGLE HOME che non sia a pagamento. È un vero peccato che solo ALEXA riesce a fare. Io sono un veterano di Goggle Home ma ultimamente sto cambiando idea.

**Reply by Pewibe** [reply]
For those who are not able to read Italien and for mikimast who has a problem to use the google trans :-) :
Good morning and happy new year.
Many days have passed and I take the liberty of asking if there is any news. I don't understand why this problem hasn't been fixed yet. I don't see the possibility of reaching a TASMOTA device with our GOOGLE HOME that is not paid for. It's a pity that only ALEXA manages to do. I'm a veteran of Goggle Home but lately I'm changing my mind.

**Reply by tyeth** [reply]
Get a NodeRed instance going. The writer of that offers paid support and free hosting, it integrates with everything but may require "some work". Or better yet pay someone to do "some work" to get the support in tasmota. Won't be instantaneous and may require more on your part than you want, but definitely get the desired result. 

Personally I've gone with sonoff (ewelink app/website), tado (cheap sockets with power monitoring), which all do google, and then some custom madness with nodered and adafruit.

In answer to your numbered items, 1) google needs to be able to send a message to a server that is always online, setup with security etc, and home-servers are beyond a lot of peoples effort levels or capabilities.
2) + 3) Sadly I'm new to developing tasmota, more of a spectator before, so can't help in this regard yet.

**Reply by sfromis** [reply]
Part of the story is that when people choose projects like Tasmota for local-only IoT, it is to avoid the need for connecting to a cloud setup. While I'm *not* saying that cloud usage would preclude local control, this philosophy does tend to limit the number of people taking on the task of developing such integrations, and submit a PR for adding to projects.

**Comment by Jason2866** [comment]
@Pewibe imho you should change the headline to something like "Feature Request: Google Home Support in Tasmota" not saying someone will jump on, but the current headline is misleading. Since currently there is no support in Tasmota for Google Home. Maybe never...

**Comment by Pewibe** [comment]
New year, new run for Google Home? 
Let me express and point out something which I might have misunderstood or other guys do not recognize the advantage: It's the difference between "compatible" and "seamless" devices/setup. "compatible" stand for the traditional way where the hardware manufacturer provides a web service and Google Home talks with them. The web service talks to the hardware and enforces the device to to more or less commands to get the user lucky. This leads to phrases "cloud service", no local service, nothing for Tasmota, etc. 
As Google acquires IoT devices manufacturing companies they develop a local and direct interface between control devices (eg. Goggle Home Mini) and acting devices (eg. a thermometer or a power plug). Using the phrase "seamless setup" of non-google smart devices they opened this development for everyone to produce compatible devices. Such devices (e.g. LEDVANCE/Sylvania SMART+) can operate direct with Google Home devices and do NOT need a web service by the manufacturer. It's up to the user to control the devices locally by Google Home (does it run really local only?) or extended with Google Assistent by speech where (today) cloud is needed.
Keep in mind that all above might be not true in details because of my semi knowledge. This is why I'm looking for cracks in this area :-) 
Could this be a starter for a future Tasmota interface to allow Google Home to discover/setup Tasmota devices SEAMLESS?

**Reply by tyeth** [reply]
I'm curious what they're using. Plenty of options but no idea. Basically like DLNA, more likely matter, but also the ESP brand has similar lan-control ([rainmaker](https://rainmaker.espressif.com/docs/local-control.html)).

**Reply by sfromis** [reply]
Another thing which may matter to this discussion is the work-in-process of Tasmota support for Matter. While I do not know how it will pan out, it does sound plausible that Google will make some sort of accommodation for Matter devices to work with however Google will integrate Google Home (Google Assistant) with Matter. Maybe that would allow for Matter ecosystem devices integration with their voice assistant without each device needing to have specific support for the talkative bot. Or maybe just provide some metadata. Obviously, this is just speculation.

**Reply by Jason2866** [reply]
Matter maybe is a way. It is in the works for Tasmota32. Do not ask when it is ready. It will take some months. A lot of work ahead... It is done when it is done :-) Probably no one will work on a solution for Google integration when there is a chance to solve somehow with Matter.

**Reply by sfromis** [reply]
If/when Tasmota gets support for Matter on ESP32, it would be plausible that ESP8266-based Tasmota devices will be able to use a device with ESP32 as a gateway to Matter too.

**Reply by tyeth** [reply]
The espressif video on matter is interesting, and their webpage is informative especially as some models are intended as border routers, but this covers the basics: https://www.androidpolice.com/matter-smart-home-standard-explained/
and we want the thread protocol for offline support. Patience, or give it 12-18months and check back :)
